app/lib/features/chat/widgets/image_message_builder.dart

```diff
@@ -37,7 +41,7 @@ class ImageMessageBuilder extends ConsumerWidget {
                 maxWidth: MediaQuery.of(context).size.width * 0.2,
               ),
               child: Image.memory(
-                data,
+                data!,
```

Comment:
Wait, what? You are changing the Provider from a `Uint8List` to a `Uint8List?`, with the explicit case that _on error_ it won't error but come back with `null` and then you have the UI do a `!` on the return value... 

Nope. I'm sorry. But there is plenty of reasons why loading the imageBinary might fail (network, IO, decryption, decoding ...) and just ignoring the error (it isn't even logged or printed!) and then going on but _hard fail_ on a `null`, that was explicitly returned isn't an acceptable way to deal with the situation. In general we need to decrease the hard-null `!` in our code base, it is mostly based on assumption of limited information and it leads to catastrophic failures with the end-user clients in cases, where - if we handled that better - we could at least given them an explanation of with an error message to understand what went wrong.  

The code was better before this change, as at least the error code was run when the image loading failed.

Reply:
I will change it. Apologies!